soc/silabs/silabs_siwx917/siwg917/nwp_init.c

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2024 Silicon Laboratories Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+
+#include "sl_wifi.h"
+#include "sl_wifi_callback_framework.h"
+#if IS_ENABLED(CONFIG_BT_SIWX917)
```

Comment:
Oh, IS_ENABLED() is designed to be used with C code:
```
   if (IS_ENABLED(CONFIG_BT_SIWX917))
```
In the case above, you can use `#ifdef` as usual.

Reply:
Yeah, the recommendation is to use `#ifdef CONFIG_...` or `#if defined(CONFIG_...)`. The latter is more convenient if you need to combine it with other conditions (e.g. other Kconfig options). In Linux it makes sense to use `IS_ENABLED()` elsewhere too, since there it covers both `=m` and `=y`, however Zephyr doesn't have a concept of building features as modules like Linux does.

Reply:
I also wanted to add that there's been recent discussions on making the references to Kconfig options consistent across the Zephyr code base, and there the consensus has been to limit `IS_ENABLED()` strictly to C code and not use it with the preprocessor.

Reply:
Noted, will go on with the simple preprocessor compilation directives.